eth/downloader/downloader.go

```diff
@@ -711,8 +720,10 @@ func (d *Downloader) findAncestor(p *peerConnection, remoteHeader *types.Header)
 		localHeight = d.blockchain.CurrentBlock().NumberU64()
 	case FastSync:
 		localHeight = d.blockchain.CurrentFastBlock().NumberU64()
-	default:
+	case LightSync:
 		localHeight = d.lightchain.CurrentHeader().Number.Uint64()
+	default:
```

Comment:
You've added these in three places, I can't say I think it's a worthwhile improvement, really. Mind reverting that and make the diff smaller?

Reply:
The reason I like this is that it removes what I would call "incidental" logic, where a reader/developer has to know (and keep in their mind) an implicit context; ie, they have to know, and think about, that there are 3 and only 3 downloader sync modes, and then read between the lines to understand that the `d.lightchain` assignment happens in the case of the downloader's `LightSync` configuration (and again further, that `LightSync` isn't the application default). IMO It's taking advantage of a terse logic pattern at the price of readability, and I would prefer explicit readability at the cost of a few lines.

Hypothetically, with the proposed change, if, say, a developer wants to try adding a 4th downloader sync mode on a given weekend, they won't have to spend any time working to understand a mysterious bug that will turn out to be the switch statement _not_ complaining about an unhandled case.

Of course, you're the owner here, and I'm willing to concede the change with no hard feelings, but just to offer my two cents.